### PR TITLE
Add more CI linting based on black, pycodestyle, and pydocstyle

### PR DESCRIPTION
**Pull request by laracroft37**
This pull request extends the linting workflow with more checks and adapts the code base for these to pass.

### CODE REVIEW

File: guibot/region.py [hunk 1]
```diff
@@ -315,13 +334,18 @@ def above(self, rrange: int = 0) -> "Region":
...
+            new_height,
+            self.dc_backend,
+            self.cv_backend,
+        )
 
     def below(self, rrange: int = 0) -> "Region":
         """
-        Obtain a region containing the previous one but enlarged
-        by a number of pixels on the lower side.
+        Enlarge the previous region by a number of pixels on the lower side.
```
Comment (pevogam) [inline guibot/region.py:348]: Same here

File: guibot/region.py [hunk 2]
```diff
@@ -333,13 +357,18 @@ def below(self, rrange: int = 0) -> "Region":
...
+            new_height,
+            self.dc_backend,
+            self.cv_backend,
+        )
 
     def left(self, rrange: int = 0) -> "Region":
         """
-        Obtain a region containing the previous one but enlarged
-        by a number of pixels on the left side.
+        Enlarge the previous region by a number of pixels on the left side..
```
Comment (pevogam) [inline guibot/region.py:371]: Same here

File: guibot/region.py [hunk 3]
```diff
@@ -356,13 +385,18 @@ def left(self, rrange: int = 0) -> "Region":
...
+            self._height,
+            self.dc_backend,
+            self.cv_backend,
+        )
 
     def right(self, rrange: int = 0) -> "Region":
         """
-        Obtain a region containing the previous one but enlarged
-        by a number of pixels on the right side.
+        Enlarge the previous region by a number of pixels on the right side.
```
Comment (pevogam) [inline guibot/region.py:399]: Same here

File: guibot/region.py [hunk 4]
```diff
@@ -479,17 +536,22 @@ def _determine_cv_backend(self, target: Target) -> "Match":
...
+            )
         target.match_settings = self.cv_backend
         return self.cv_backend
 
     def sample(self, target: str | Target) -> float:
         """
-        Sample the similarity between a target and the screen,
+        Sample the similarity between a target and the screen.
+
         i.e. an empirical probability that the target is on the screen.
```
Comment (pevogam) [inline guibot/region.py:555]: Similarity here means an empirical...

File: guibot/region.py [hunk 5]
```diff
@@ -511,8 +573,7 @@ def sample(self, target: str | Target) -> float:
 
     def exists(self, target: str | Target, timeout: int = 0) -> "Match | None":
         """
-        Check if a target exists on the screen using the matching
-        success as a threshold for the existence.
+        Check if a target exists on the screen using the matching success as a threshold.
```
Comment (pevogam) [inline guibot/region.py:576]: ...using similarity as a threshold.

File: guibot/region.py [hunk 6]
```diff
@@ -541,8 +601,9 @@ def wait(self, target: str | Target, timeout: int = 30) -> "Match":
 
     def wait_vanish(self, target: str | Target, timeout: int = 30) -> "Region":
         """
-        Wait for a target to disappear (be unmatched, i.e. matched
-        without success) with a given timeout as failing tolerance.
+        Wait for a target to disappear with a given timeout as failing tolerance.
```
Comment (pevogam) [inline guibot/region.py:604]: ...disappear (be unmached) with...

File: guibot/region.py [hunk 7]
```diff
@@ -541,8 +601,9 @@ def wait(self, target: str | Target, timeout: int = 30) -> "Match":
...
+
+        (Be unmatched, i.e. matched without success)
```
Comment (pevogam) [inline guibot/region.py:606]: This should be removed then

File: guibot/region.py [hunk 8]
```diff
@@ -608,11 +672,13 @@ def hover(self, target_or_location: "Match | Location | str | Target") -> "Match
...
-              modifiers: list[str] = None) -> "Match | None":
+    def click(
+        self,
+        target_or_location: "Match | Location | str | Target",
+        modifiers: list[str] = None,
+    ) -> "Match | None":
         """
-        Click on a target or location using the left mouse button and
-        optionally holding special keys.
+        Click on a target or location using the left mouse button and special keys.
```
Comment (pevogam) [inline guibot/region.py:681]: Click on a target or location using the left mouse button.

Optionally we can hold some special keys.

File: guibot/region.py [hunk 9]
```diff
@@ -630,11 +696,13 @@ def click(self, target_or_location: "Match | Location | str | Target",
...
-                    modifiers: list[str] = None) -> "Match | None":
+    def right_click(
+        self,
+        target_or_location: "Match | Location | str | Target",
+        modifiers: list[str] = None,
+    ) -> "Match | None":
         """
-        Click on a target or location using the right mouse button and
-        optionally holding special keys.
+        Click on a target or location using the right mouse button and special keys.
```
Comment (pevogam) [inline guibot/region.py:705]: Same

File: guibot/region.py [hunk 10]
```diff
@@ -645,11 +713,13 @@ def right_click(self, target_or_location: "Match | Location | str | Target",
...
-                     modifiers: list[str] = None) -> "Match | None":
+    def middle_click(
+        self,
+        target_or_location: "Match | Location | str | Target",
+        modifiers: list[str] = None,
+    ) -> "Match | None":
         """
-        Click on a target or location using the middle mouse button and
-        optionally holding special keys.
+        Click on a target or location using the middle mouse button and special keys.
```
Comment (pevogam) [inline guibot/region.py:722]: Same

File: guibot/region.py [hunk 11]
```diff
@@ -660,11 +730,13 @@ def middle_click(self, target_or_location: "Match | Location | str | Target",
...
-                     modifiers: list[str] = None) -> "Match | None":
+    def double_click(
+        self,
+        target_or_location: "Match | Location | str | Target",
+        modifiers: list[str] = None,
+    ) -> "Match | None":
         """
-        Double click on a target or location using the left mouse button
-        and optionally holding special keys.
+        Double click on a target or location using the left mouse button special keys.
```
Comment (pevogam) [inline guibot/region.py:739]: Same

File: guibot/region.py [hunk 12]
```diff
@@ -675,11 +747,14 @@ def double_click(self, target_or_location: "Match | Location | str | Target",
...
+    def multi_click(
+        self,
+        target_or_location: "Match | Location | str | Target",
+        count: int = 3,
+        modifiers: list[str] = None,
+    ) -> "Match | None":
         """
-        Click N times on a target or location using the left mouse button
-        and optionally holding special keys.
+        Click N times on a target or location using the left mouse button and special keys.
```
Comment (pevogam) [inline guibot/region.py:757]: Same

File: guibot/region.py [hunk 13]
```diff
@@ -740,18 +825,22 @@ def click_vanish(self, click_image_or_location: Image | Location,
...
+        self,
+        anchor: str | Target,
+        index: int = 0,
+        find_number: int = 3,
+        timeout: int = 10,
+    ) -> "Match":
         """
-        Find all instances of an anchor image and click on the one with the
-        desired index given that they are horizontally then vertically sorted.
+        Find and click on a specific instance of an anchor image, given their horizontal and vertical sorting.
```
Comment (pevogam) [inline guibot/region.py:836]: Find and click on a specific instance of an anchor image indexed by horizontal and vertical sorting.

File: guibot/region.py [hunk 14]
```diff
@@ -1161,13 +1312,22 @@ def fill_at(self, anchor: "Match | Location | Target | str",
...
+        ret_flag: bool = True,
+        mark_clicks: int = 1,
+        tries: int = 3,
+    ) -> "Region":
         """
-        Select an option at a dropdown list using either an integer index
-        or an option image if the order cannot be easily inferred.
+        Select an option at a dropdown list using either an integer index.
+
+        Or select an option image if the order cannot be easily inferred.
```
Comment (pevogam) [inline guibot/region.py:1330]: Select an option at a dropdown list using an index or an image.

The caller can use either integer index or an option image if the order cannot be easily inferred.

**Review by pevogam — CHANGES_REQUESTED** [review]
Great conversion, mostly some docstrings and wording there to improve upon.

File: guibot/calibrator.py [hunk 15]
```diff
@@ -37,29 +37,33 @@
 from .location import Location
 
 import logging
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/calibrator.py:39]: Done

File: guibot/calibrator.py [hunk 16]
```diff
@@ -138,33 +158,53 @@ def backend_tuples(category_list: list[str], finder: Finder) -> Generator[tuple[
...
+        random_starts: int = 1,
+        uniform: bool = False,
+        calibration: bool = True,
+        max_attempts: int = 3,
+        **kwargs: dict[str, type]
+    ) -> float:
         """
-        Search for the best match configuration for a given needle and haystack
-        using calibration from random initial conditions.
+        Find the best match configuration for a given needle and haystack using random calibration.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/calibrator.py:207]: Done

File: guibot/calibrator.py [hunk 17]
```diff
@@ -220,14 +275,21 @@ def search(self, finder: Finder, random_starts: int = 1, uniform: bool = False,
...
         return 1.0 - best_error
 
-    def calibrate(self, finder: Finder, max_attempts: int = 3, **kwargs: dict[str, type]) -> float:
+    def calibrate(
+        self, finder: Finder, max_attempts: int = 3, **kwargs: dict[str, type]
+    ) -> float:
         """
-        Calibrate the available match configuration for a given needle
-        and haystack minimizing the matchign error.
+        Calibrate the match configuration for a given needle and haystack to minimize the matching error..
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/calibrator.py:292]: Done

File: guibot/config.py [hunk 18]
```diff
@@ -553,9 +623,9 @@ class GlobalConfig(object, metaclass=GlobalConfig):  # type: ignore
 
 class TemporaryConfig(object):
     """
-    Proxies a GlobalConfig instance extending it to add context
-    support, such that once this context ends the changes to the
-    wrapped config object are restored.
+    Proxies a GlobalConfig instance extending it to add context support.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/config.py:626]: Done

File: guibot/config.py [hunk 19]
```diff
@@ -553,9 +623,9 @@ class GlobalConfig(object, metaclass=GlobalConfig):  # type: ignore
...
+
+    Such that once this context ends the changes to the wrapped config object are restored.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/config.py:628]: Done

File: guibot/config.py [hunk 20]
```diff
@@ -606,9 +680,9 @@ def __exit__(self, *_: tuple[type, ...]) -> None:
 
 class LocalConfig(object):
     """
-    Container for the configuration of all display control and
-    computer vision backends, responsible for making them behave
-    according to the selected parameters as well as for providing
+    Container for the configuration of all display control and computer vision backends.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/config.py:683]: Done

File: guibot/config.py [hunk 21]
```diff
@@ -606,9 +680,9 @@ def __exit__(self, *_: tuple[type, ...]) -> None:
...
+
+    Responsible for making them behave according to the selected parameters as well as for providing
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/config.py:685]: Done

File: guibot/controller.py [hunk 22]
```diff
@@ -40,15 +40,21 @@
...
+    "VNCDoToolController",
+    "PyAutoGUIController",
+]
 
 
 class Controller(LocalConfig):
     """
-    Screen control backend, responsible for performing desktop operations
-    like mouse clicking, key pressing, text typing, etc.
+    Screen control backend, responsible for performing desktop operations.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/controller.py:55]: Done

File: guibot/controller.py [hunk 23]
```diff
@@ -40,15 +40,21 @@
...
+
+    Like mouse clicking, key pressing, text typing, etc.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/controller.py:57]: Done

File: guibot/controller.py [hunk 24]
```diff
@@ -128,47 +143,60 @@ def get_mouse_location(self) -> Location:
...
         log.log(9, "%s %s\n", category, self.params[category])
 
-    def configure_backend(self, backend: str = None, category: str = "control",
-                          reset: bool = False) -> None:
+    def configure_backend(
+        self, backend: str = None, category: str = "control", reset: bool = False
+    ) -> None:
         """
-        Custom implementation of the base method.
+        Customizeize implementation of the base method.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/controller.py:175]: Done

File: guibot/controller.py [hunk 25]
```diff
@@ -128,47 +143,60 @@ def get_mouse_location(self) -> Location:
...
+            )
 
-    def synchronize_backend(self, backend: str = None, category: str = "control",
-                            reset: bool = False) -> None:
+    def synchronize_backend(
+        self, backend: str = None, category: str = "control", reset: bool = False
+    ) -> None:
         """
-        Custom implementation of the base method.
+        Customizeize implementation of the base method.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/controller.py:199]: Done

File: guibot/controller.py [hunk 26]
```diff
@@ -308,13 +352,16 @@ def keys_type(self, text: list[str] | str, modifiers: list[str] = None) -> None:
...
+        )
 
 
 class AutoPyController(Controller):
     """
-    Screen control backend implemented through AutoPy which is a small
-    python library portable to Windows and Linux operating systems.
+    Screen control backend implemented through AutoPy.
+
+    A small python library portable to Windows and Linux operating systems.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/controller.py:364]: Done

File: guibot/controller.py [hunk 27]
```diff
@@ -379,41 +441,53 @@ def __synchronize_backend(self, backend: str = None, category: str = "autopy",
...
         self._mousemap = inputmap.AutoPyMouseButton()
 
-    def synchronize_backend(self, backend: str = None, category: str = "autopy",
-                          reset: bool = False) -> None:
+    def synchronize_backend(
+        self, backend: str = None, category: str = "autopy", reset: bool = False
+    ) -> None:
         """
-        Custom implementation of the base method.
+        Customize implementation of the base method.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/controller.py:448]: Done

File: guibot/controller.py [hunk 28]
```diff
@@ -379,41 +441,53 @@ def __synchronize_backend(self, backend: str = None, category: str = "autopy",
...
 
         See base method for details.
         """
         self.__synchronize_backend(backend, category, reset)
 
     def capture_screen(self, *args: "list[int] | Region | None") -> Image:
         """
-        Custom implementation of the base method.
+        Customize implementation of the base method.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/controller.py:456]: Done

File: guibot/controller.py [hunk 29]
```diff
@@ -379,41 +441,53 @@ def __synchronize_backend(self, backend: str = None, category: str = "autopy",
...
         with PIL.Image.open(filename) as f:
-            pil_image = f.convert('RGB')
+            pil_image = f.convert("RGB")
         os.unlink(filename)
         return Image("", pil_image)
 
     def mouse_move(self, location: Location, smooth: bool = True) -> None:
         """
-        Custom implementation of the base method.
+        Customize implementation of the base method.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/controller.py:490]: Done

File: guibot/controller.py [hunk 30]
```diff
@@ -424,10 +498,11 @@ def mouse_move(self, location: Location, smooth: bool = True) -> None:
...
         self._pointer = location
 
-    def mouse_click(self, button: int = None, count: int = 1,
-                    modifiers: list[str] = None) -> None:
+    def mouse_click(
+        self, button: int = None, count: int = 1, modifiers: list[str] = None
+    ) -> None:
         """
-        Custom implementation of the base method.
+        Customize implementation of the base method.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/controller.py:505]: Done

File: guibot/controller.py [hunk 31]
```diff
@@ -447,23 +522,23 @@ def mouse_click(self, button: int = None, count: int = 1,
 
     def mouse_down(self, button: int) -> None:
         """
-        Custom implementation of the base method.
+        Customize implementation of the base method.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/controller.py:525]: Done

File: guibot/controller.py [hunk 32]
```diff
@@ -627,52 +728,53 @@ def mouse_click(self, button: int = None, count: int = 1,
...
             self.keys_toggle(modifiers, False)
 
 
 class VNCDoToolController(Controller):
     """
-    Screen control backend implemented through the VNCDoTool client and
-    thus portable to any guest OS that is accessible through a VNC/RFB protocol.
+    Screen control backend implemented through the VNCDoTool client.
+
+    Portable to any guest OS that is accessible through a VNC/RFB protocol.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/controller.py:777]: Done

File: guibot/controller.py [hunk 33]
```diff
@@ -867,8 +991,9 @@ def keys_type(self, text: list[str] | str, modifiers: list [str] = None) -> None
 
 class PyAutoGUIController(Controller):
     """
-    Screen control backend implemented through PyAutoGUI which is a python
-    library portable to MacOS, Windows, and Linux operating systems.
+    Screen control backend implemented through PyAutoGUI.
+
+    Which is a python library portable to MacOS, Windows, and Linux operating systems.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/controller.py:996]: Done

File: guibot/finder.py [hunk 34]
```diff
@@ -105,20 +121,32 @@ def __init__(self, value: bool | int | float | str | None,
...
+                self.tolerance,
+                self.fixed,
+                self.enumerated,
+            )
+        )
 
     def __eq__(self, other: "CVParameter") -> bool:
         """
-        Custom implementation for equality check.
+        Customize implementation for equality check.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/finder.py:149]: Done

File: guibot/finder.py [hunk 35]
```diff
@@ -322,55 +368,68 @@ def __init__(self, configure: bool = True, synchronize: bool = True) -> None:
...
         log.log(9, "%s %s\n", category, self.params[category])
 
-    def configure_backend(self, backend: str = None, category: str = "find",
-                          reset: bool = False) -> None:
+    def configure_backend(
+        self, backend: str = None, category: str = "find", reset: bool = False
+    ) -> None:
         """
-        Custom implementation of the base method.
+        Customize implementation of the base method.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/finder.py:398]: Done

File: guibot/finder.py [hunk 36]
```diff
@@ -322,55 +368,68 @@ def __init__(self, configure: bool = True, synchronize: bool = True) -> None:
...
         See base method for details.
         """
         self.__synchronize_backend(backend, category, reset)
 
     def can_calibrate(self, category: str, mark: bool) -> None:
         """
-        Fix the parameters for a given category backend algorithm,
+        Fix the parameters for a given category backend algorithm.
+
         i.e. disallow the calibrator to change them.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/finder.py:433]: Done

File: guibot/finder.py [hunk 37]
```diff
@@ -582,37 +661,48 @@ def __init__(self, configure: bool = True, synchronize: bool = True) -> None:
...
             self.__configure(reset=True)
 
-    def __configure_backend(self, backend: str = None, category: str = "contour",
-                            reset: bool = False) -> None:
+    def __configure_backend(
+        self, backend: str = None, category: str = "contour", reset: bool = False
+    ) -> None:
         """
-        Custom implementation of the base method.
+        Customize implementation of the base method.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/finder.py:668]: Done

File: guibot/finder.py [hunk 38]
```diff
@@ -806,7 +972,7 @@ def _extract_contours(self, countours_image: "Matlike", log: bool = False) -> "l
 
     def log(self, lvl: int) -> None:
         """
-        Custom implementation of the base method.
+        Customize implementation of the base method.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/finder.py:975]: Done

File: guibot/finder.py [hunk 39]
```diff
@@ -1443,36 +1792,52 @@ def _detect_features(self, ngray: int, hgray: int, detect: str,
...
         EXTRA DOCSTRING: Feature matching backend - matching stage (2).
 
         Match two sets of keypoints based on their descriptors.
         """
+
         def ratio_test(matches: list[Any]) -> list[Any]:
             """
-            The ratio test checks the first and second best match. If their
-            ratio is close to 1.0, there are both good candidates for the
+            Ratio test checks the first and second best match.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/finder.py:1838]: Done

File: guibot/finder.py [hunk 40]
```diff
@@ -1495,13 +1863,15 @@ def ratio_test(matches: list[Any]) -> list[Any]:
 
         def symmetry_test(nmatches: list[Any], hmatches: list[Any]) -> list[Any]:
             """
-            Refines the matches with a symmetry test which extracts
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/finder.py:1498]: Done

File: guibot/guibot_proxy.py [hunk 41]
```diff
@@ -45,10 +44,11 @@
...
 
 
-def serialize_custom_error(class_obj: type) -> dict[str, "str | getset_descriptor | dictproxy"]:
+def serialize_custom_error(
+    class_obj: type,
+) -> dict[str, "str | getset_descriptor | dictproxy"]:
     """
-    Serialization method for the :py:class:`errors.UnsupportedBackendError`
-    which was chosen just as a sample.
+    Serialize method for the :py:class:`errors.UnsupportedBackendError` which was chosen just as a sample.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/guibot_proxy.py:51]: Done

File: guibot/guibot_proxy.py [hunk 42]
```diff
@@ -62,21 +62,23 @@ def serialize_custom_error(class_obj: type) -> dict[str, "str | getset_descripto
 
 def register_exception_serialization() -> None:
     """
-    We put here any exceptions that are too complicated for the default serialization
-    and define their serialization methods.
+    We put exceptions that are too complicated for the default serialization and define their serialization methods.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/guibot_proxy.py:65]: Done

File: guibot/guibot_proxy.py [hunk 43]
```diff
@@ -62,21 +62,23 @@ def serialize_custom_error(class_obj: type) -> dict[str, "str | getset_descripto
...
+        pyro.util.SerializerBase.register_class_to_dict(
+            exception, serialize_custom_error
+        )
 
 
 class GuiBotProxy(GuiBot):
     """
-    The proxy guibot object is just a wrapper around the actual guibot
-    object that takes care of returning easily serializable PyRO proxy objects
+    The proxy guibot object is just a wrapper around the actual guibot object.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/guibot_proxy.py:79]: Done

File: guibot/region.py [hunk 44]
```diff
@@ -43,17 +42,26 @@
...
-log = logging.getLogger('guibot.region')
+
+log = logging.getLogger("guibot.region")
 
 
 class Region(object):
     """
-    Region of the screen supporting vertex and nearby region selection,
-    validation of expected images, and mouse and keyboard control.
+    Region of the screen supporting vertex and nearby region selection.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/region.py:51]: Done

File: guibot/region.py [hunk 45]
```diff
@@ -292,13 +311,13 @@ def nearby(self, rrange: int = 50) -> "Region":
...
-                      self.dc_backend, self.cv_backend)
+        return Region(
+            new_xpos, new_ypos, new_width, new_height, self.dc_backend, self.cv_backend
+        )
 
     def above(self, rrange: int = 0) -> "Region":
         """
-        Obtain a region containing the previous one but enlarged
-        by a number of pixels on the upper side.
+        Enlarge the previous region by a number of pixels on the upper side.
```
[the remark this answers is not in the document]
Comment (laracroft37) [reply guibot/region.py:320]: Done

File: guibot/region.py (hunk 1 above)
Comment (laracroft37) [reply guibot/region.py:348]: Done

File: guibot/region.py (hunk 2 above)
Comment (laracroft37) [reply guibot/region.py:371]: Done

File: guibot/region.py (hunk 3 above)
Comment (laracroft37) [reply guibot/region.py:399]: Done

File: guibot/region.py (hunk 4 above)
Comment (laracroft37) [reply guibot/region.py:555]: Done

File: guibot/region.py (hunk 5 above)
Comment (laracroft37) [reply guibot/region.py:576]: Done

File: guibot/region.py (hunk 6 above)
Comment (laracroft37) [reply guibot/region.py:604]: Done

File: guibot/region.py (hunk 8 above)
Comment (laracroft37) [reply guibot/region.py:681]: Done

Comment (laracroft37) [reply guibot/region.py:681]: Done

File: guibot/region.py (hunk 9 above)
Comment (laracroft37) [reply guibot/region.py:705]: Done

File: guibot/region.py (hunk 10 above)
Comment (laracroft37) [reply guibot/region.py:722]: Done

File: guibot/region.py (hunk 11 above)
Comment (laracroft37) [reply guibot/region.py:739]: Done

File: guibot/region.py (hunk 13 above)
Comment (laracroft37) [reply guibot/region.py:836]: Done

File: guibot/region.py (hunk 14 above)
Comment (laracroft37) [reply guibot/region.py:1330]: Done

**Review by pevogam — APPROVED** [review]
LGTM and thanks!

**Pull request merged**